### PR TITLE
Separate manageClient into more files for readability

### PR DESCRIPTION
**Pull request by gormal**
### Motivation

ManagementClient class has become disorganized and confusing, so it's split into many files for readability.

### CODE REVIEW

**Review by MiroKentico — COMMENTED** [review]
I would prefer file `ManagementClient.Taxonomy.cs` renamed to more `ManagementClient.TaxonomyGroup.cs`. From what I saw, this name is used in documentation and also in model classess.

File: Kentico.Kontent.Management/ManagementClient.Workflow.cs [hunk 1]
```diff
@@ -30,4 +31,113 @@ public async Task DeleteWorkflowAsync(Reference identifier)
...
+        }
+
+        if (workflowStepIdentifier == null)
+        {
+            throw new ArgumentNullException(nameof(workflowStepIdentifier));
+        }
+
+        var endpointUrl = _urlBuilder.BuildWorkflowChangeUrl(identifier);
+
+        await _actionInvoker.InvokeMethodAsync(endpointUrl, HttpMethod.Put);
```
[the remark this answers is not in the document]
Comment (gormal) [reply Kentico.Kontent.Management/ManagementClient.Workflow.cs]: There were changes made to master after the creation of this PR, it should be ok now.

**Comment by gormal** [comment]
> I would prefer file `ManagementClient.Taxonomy.cs` renamed to more `ManagementClient.TaxonomyGroup.cs`. From what I saw, this name is used in documentation and also in model classess.

yeah, good point.

File: Kentico.Kontent.Management/ManagementClient.ProjectUser.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
+﻿using Kentico.Kontent.Management.Models.Shared;
+using Kentico.Kontent.Management.Models.Users;
+using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace Kentico.Kontent.Management;
+
+public partial class ManagementClient : IManagementClient
```
[the remark this answers is not in the document]
Comment (gormal) [reply Kentico.Kontent.Management/ManagementClient.ProjectUser.cs]: Yeah, it makes the code more concise.

File: Kentico.Kontent.Management/ManagementClient.Workflow.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (MiroKentico) [reply Kentico.Kontent.Management/ManagementClient.Workflow.cs]: I see `workflowStepIdentifier` parameter there now, so I suppose it was really just some not yet reflected changes from master.

**Comment by MiroKentico** [comment]
Thanks for reflecting my comments, looks good now. Approved

**Review by MiroKentico — APPROVED** [review]

**Pull request merged**